src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchClient.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+@Component
+class PrisonerSearchClient(private val prisonerSearchWebClient: WebClient) {
+  fun matchPrisoners(prisonerSearchRequest: PrisonerSearchRequest): MatchPrisonersResponse {
+    return prisonerSearchWebClient
+      .post()
+      .uri("/match-prisoners")
+      .accept(MediaType.APPLICATION_JSON)
+      .bodyValue(
+        mapOf(
+          "nomsNumber" to prisonerSearchRequest.prisonNumber,
+          "firstName" to prisonerSearchRequest.firstName,
+          "lastName" to prisonerSearchRequest.lastName,
+          "dateOfBirth" to prisonerSearchRequest.dob?.format(DateTimeFormatter.BASIC_ISO_DATE)
+        )
+      )
+      .retrieve()
+      .bodyToMono<MatchPrisonersResponse>()
+      .block()!!
```

Comment:
Not for this PR, but I was wondering if we could run several prisoner searches in parallel using coroutines? We don't care about the results yet so could even take them completely off the main thread.